### ISSUE TITLE
Suspicious state from syntax checker r-lintr: Flycheck checker r-lintr returned 1, but its output contained no errors

### ISSUE THREAD

**Issue by solmos** [flycheck/flycheck#1991]
## Bug description
I get the following error anytime I am typing inside the argument parentheses in a function definition in an R script:

```
Suspicious state from syntax checker r-lintr: Flycheck checker r-lintr returned 1, but its output contained no errors: Error in rep.int(character, length) : invalid 'times' value
Calls: <Anonymous> ... print.lint -> cat -> highlight_string -> fill_with -> paste0
In addition: Warning messages:
1: In readLines(obj) :
  incomplete final line found on '/private/var/folders/gx/vdqbfzvs66b1swy4qv7pch7c0000gp/T/flycheckxgcutS/test.R'
2: In readLines(obj) :
  incomplete final line found on '/private/var/folders/gx/vdqbfzvs66b1swy4qv7pch7c0000gp/T/flycheckxgcutS/test.R'
Execution halted

Try installing a more recent version of r-lintr, and please open a bug report if the issue persists in the latest release.  Thanks!
```

## Steps to reproduce

1. Open file `test.R`

2. Type:

```r
foo <- function()
```

4. Place cursor (in insert mode using evil-mode) inside the parentheses.

## Screenshots

<img width="1497" alt="Screenshot 2023-01-05 at 13 39 03" src="https://user-images.githubusercontent.com/11445933/210783976-f437ba7b-cd0e-4878-9e76-7ac2c81f6838.png">

## System configuration

```
Syntax checkers for buffer test.R in ess-r-mode:

First checker to run:

  r-lintr
    - may enable:    yes
    - may run:       t
    - executable:    Found at /usr/local/bin/R
    - lintr library: present

Flycheck Mode is enabled. Use SPC u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot
Emacs version:    28.2
System:           aarch64-apple-darwin21.6.0
Window system:    mac
```

Emacs configuration: Doom Emacs

**Comment by teotenn** [comment]
I was having same problem as you, without the middle part of `In addition: Warning messages:`

Try checking the syntax of your `.lintr` file. Flycheck is very special for the syntax. For example in my case I was trying something like this

```
(setq tt/lintr-linters
      "linters_with_defaults(
         line_length_linter = line_length_linter(120)
       )"
 )

(setq flycheck-lintr-linters tt/lintr-linters))
```

Something that with Flymake kind of makes sense and works. But Flycheck keeps sending me the same error as you got. So changed it for `(setq flycheck-lintr-linters "linters_with_defaults(line_length_linter = line_length_linter(120))"))` and now Flycheck is working.

**For the mainteiners**, is there a reason why, when I set a variable with spaces flycheck does not recognize it? Flymake had no problems recognizing it.

**Comment by bbatsov** [comment]
Closing: this is a bug in the `lintr` R package crashing on incomplete code, not a flycheck issue. Please report upstream.
